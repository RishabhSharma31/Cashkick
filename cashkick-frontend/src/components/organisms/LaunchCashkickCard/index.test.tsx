import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import CashKickCard from './index';

describe('CashKickCard Component', () => {
  const mockOnClick = jest.fn();

  const setup = () =>
    render(
      <CashKickCard
        title="Launch a new Cash Kick"
        description="You have up to {amount} available for a new cash advance."
        amount="$880,000.00"
        buttonText="New Cash Kick"
        onButtonClick={mockOnClick}
      />
    );

  it('renders the title', () => {
    setup();
    expect(screen.getByText('Launch a new Cash Kick')).toBeInTheDocument();
  });

  it('renders the interpolated description with amount', () => {
    setup();
    expect(
      screen.getByText('You have up to $880,000.00 available for a new cash advance.')
    ).toBeInTheDocument();
  });

  it('renders the button with correct text', () => {
    setup();
    expect(screen.getByRole('button', { name: 'New Cash Kick' })).toBeInTheDocument();
  });

  it('calls onButtonClick when the button is clicked', () => {
    setup();
    const button = screen.getByRole('button', { name: 'New Cash Kick' });
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches the snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
