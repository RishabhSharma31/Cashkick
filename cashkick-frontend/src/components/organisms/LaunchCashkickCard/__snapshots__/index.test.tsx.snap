// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CashKickCard Component matches the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1154xrt"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1m2xbl4-MuiTypography-root"
    >
      Launch a new Cash Kick
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1buac67-MuiTypography-root"
    >
      You have up to $880,000.00 available for a new cash advance.
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1mqce9w-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      New Cash Kick
    </button>
  </div>
</div>
`;
